### PR TITLE
Introduce retryable errors to Kafka.

### PR DESCRIPTION
**Pull request by PremyslCerny**
@mejroslav Introducing retryable errors to kafka source (there is only one now).

### CODE REVIEW

File: bspump/kafka/source.py [hunk 1]
```diff
@@ -260,8 +271,24 @@ def poll_kafka(self, consumer, storage):
...
+				# Handle retryable errors
+				# like UNKNOWN_TOPIC_OR_PART
+				if kafka_error.code() in self.RetryableErrors:
+
+					if kafka_error_retry_attempts < self.ErrorsMaxRetryAttemps:
+						kafka_error_retry_attempts += 1
+						time.sleep(self.SleepOnRetryableError)
+						continue
+
+					L.warning("The following error occurred while polling for messages: '{}'.".format(m.error()))
```
Comment (mejroslav) [inline bspump/kafka/source.py:287]: Can you shorten this a little bit?
E.g.
```python
L.warning("Cannot poll messages from topic: {}".format(m.error()), struct_data={"topic": topic, "id": self.Id})
```

File: bspump/kafka/source.py [hunk 2]
```diff
@@ -108,6 +110,14 @@ def __init__(self, app, pipeline, connection, id=None, config=None):
 
 		# Sleep time after an error
 		self.SleepOnError = float(self.Config.pop("sleep_on_error", 3.0))
+		self.SleepOnRetryableError = float(self.Config.pop("sleep_on_retryable_error", 60.0))
+
+		# Retryable errors retry attempts
+		self.ErrorsMaxRetryAttemps = int(self.Config.pop("errors_max_retry_attempts", 100))
```
Comment (mejroslav) [inline bspump/kafka/source.py:116]: With this set, you get the first warning after 6000 seconds? Shouldn't you wait for example 1 second and display the warning after 1 minute?

**Review by mejroslav — CHANGES_REQUESTED** [review]
See the comment.

**Review by ateska — APPROVED** [review]

File: bspump/kafka/source.py (hunk 2 above)
Comment (PremyslCerny) [reply bspump/kafka/source.py:116]: I don't think the warning is relevant to the user honestly.

Comment (PremyslCerny) [reply bspump/kafka/source.py:116]: Let us see if there are more retryable errors. If I don't recreate the consumer, the error is seen only once either way.

**Pull request merged**